File: 0766-toeplitz-matrix/0766-toeplitz-matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        r = 0
        c = 0
        R = len(matrix)
        C = len(matrix[0])
        
        while r < R and c < C:
            val = matrix[r][c]
            i, j = r, c
            
            while i < R and j < C:
                if matrix[i][j] != val:
                    return False
                i += 1
                j += 1
            c += 1
        
        r = 1
        c = 0
        while r < R and c < C:
            val = matrix[r][c]
            i, j = r, c
            
            while i < R and j < C:
                if matrix[i][j] != val:
                    return False
                i += 1
                j += 1
            r += 1
        return True
            
        

File: 0766-toeplitz-matrix/0766-toeplitz-matrix.js
/**
 * @param {number[][]} matrix
 * @return {boolean}
 */
var isToeplitzMatrix = function(matrix) {
    let R = matrix.length;
    let C = matrix[0].length;
    
    function isDiagonal(r, c) {
        let val = matrix[r][c];
        
        while (r < R && c < C) {
            if (matrix[r][c] !== val) {
                return false;
            }
            r++;
            c++;
        }
        return true;
    }
    
    
    for (let c = 0; c < C; c++) {
        if (!isDiagonal(0, c)) {
            return false;
        }
    }
    
    for (let r = 1; r < R; r++) {
        if (!isDiagonal(r, 0)) {
            return false;
        }
    }
    
    return true;
    
};